pybamm/parameters/parameter_values.py

```diff
@@ -647,7 +647,9 @@ def _process_symbol(self, symbol):
             new_symbol._scale = self.process_symbol(symbol.scale)
             reference = self.process_symbol(symbol.reference)
             if isinstance(reference, pybamm.Vector):
-                reference = pybamm.Scalar(float(reference.evaluate()))
+                # address numpy 1.25 deprecation warning: array should have ndim=0
+                # before conversion
+                reference = pybamm.Scalar((reference.evaluate()).item())
```

Comment:
Here, and in other places too, [`np.ndarray.item()`](https://numpy.org/doc/stable/reference/generated/numpy.ndarray.item.html) is preferred to direct indexing with `[0]` – because it is faster: https://stackoverflow.com/questions/62491166/numpy-index-access-vs-numpy-array-item-performance

and `item()` _looks like_ it is faster than `item(0)`:
```python
import numpy as np
# array with ndim = 1
x = np.array([1])
```

```
%timeit y = x.item()
36.7 ns ± 0.0161 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
```
compared to
```
%timeit y = x.item()
51.2 ns ± 0.0139 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
```